Source/SelfService/Web/microservice/microserviceView/configuration/environmentVariablesSection/environmentVariableSection.tsx

```diff
@@ -0,0 +1,288 @@
+        window.open(href, '_blank');
+    };
+
+    const handleEnvVariableDownload = () => {
+        const configMapName = `${configMapPrefix}-env-variables`;
+        const href = `${getServerUrlPrefix()}/live/application/${applicationId}/configmap/${configMapName}?download=1&fileType=yaml`;
+        enqueueSnackbar(`'${configMapName}.yaml' has been downloaded.`);
+        window.open(href, '_blank');
+    };
+
+    const noEnvVariables = envVariableTableRows.length === 0;
+    const noSecretEnvVariables = envVariableTableRows.filter(envVariable => envVariable.isSecret).length === 0;
+    const noPublicEnvVariables = envVariableTableRows.filter(envVariable => !envVariable.isSecret).length === 0;
+
+    return (
+        <>
+            <Accordion
+                id='environment-variables'
+                title='Environment Variables'
+                defaultExpanded
+            >
+                <Box sx={{ mb: 2.875, button: { 'mr': 6.25, '&:last-of-type': { mr: 0 } } }}>
+                    <Button
+                        variant='text'
+                        label='Add Variable'
+                        startWithIcon={<AddCircle />}
+                        disabled={disableAddButton}
+                        onClick={handleEnvVariableAdd}
+                    />
+                    <Button
+                        variant='text'
+                        label='Delete Variable(s)'
+                        disabled={!selectedRowIds.length || noEnvVariables}
+                        startWithIcon={<DeleteRounded />}
+                        onClick={handleEnvVariableDelete}
+                    />
+                    <Button
+                        variant='text'
+                        label='Download secret env-variables yaml'
+                        disabled={noSecretEnvVariables}
+                        startWithIcon={<DownloadRounded />}
+                        onClick={handleSecretEnvVariableDownload}
+                    />
+                    <Button
+                        variant='text'
+                        label='Download env-variables yaml'
+                        disabled={noPublicEnvVariables}
+                        startWithIcon={<DownloadRounded />}
+                        onClick={handleEnvVariableDownload}
+                    />
+                </Box>
+
+                <RestartInfoBox name={microserviceName} isAlertBoxOpen={restartInfoBoxIsOpen} handleDismiss={() => setRestartInfoBoxIsOpen(false)} />
+
+                {noEnvVariables ?
+                    <EmptyDataTable
+                        title='No environment variables yet...'
+                        description={`To add your first environment variable, select 'add variable'. Provide a name, value and set its secrecy.`}
+                        buttonText='Add Variable'
+                        handleOnClick={handleEnvVariableAdd} // TODO: Sometimes throws error when clicked
```

Comment:
This TODO seems interesting 😅 - any thoughts into why it throws error sometimes?